compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/Dialog.desktop.kt

```diff
@@ -341,6 +341,7 @@ fun DialogWindow(
         LocalWindowExceptionHandlerFactory.current
     )
     val parentScene = LocalComposeScene.current
+    val layoutDirection = LocalLayoutDirection.current
```

Comment:
> This is currently failing because the LocalLayoutDirection's default provider is

We can use the Locale.getDefault() for setting the default direction. Differentiating it on Window/Dialog level usually doesn't make sense.

Reply:
Do you mean to provide it in `awaitApplication`? We can't provide it like this
```
val LocalLayoutDirection = staticCompositionLocalOf<LayoutDirection> {
    Locale.current.layoutDirection
}
```
because that's common code.

Reply:
We can provide Local on the `application` level, no need to provide it the defenition

Reply:
Understood. I'll add it to `awaitApplication`.